docs/extensions/code-review-tool-attestations.md

```diff
@@ -0,0 +1,114 @@
+# Code Review Tool Attestations
+
+Last Modified: October 22, 2024
+
+gittuf's [reference authorization
+attestations](/docs/design-document.md#reference-authorization) can be used to
+meet a threshold of approvals for a change in a Git repository. Here, every
+approver signs the attestation themselves, minimizing the trusted entities
+during verification, but this has usability concerns. To balance usability and
+security, this extension proposes the use of special attestations created by
+popular code review tools.
+
+## Attestation Schema
+
+The attestation is similar to the reference authorization, but also includes
+information indicating the approvers.
+
+```
+TargetRef    string
+FromTargetID string
+ToTargetID   string
+Approvers    []string
+```
+
+The attestation must indicate the code review tool via the predicate type, and
+it must be signed by the code review tool or a trusted integration with the
+tool. See [verification](#verification) for how these attestations are
+authenticated.
+
+## Policy
+
+gittuf policy must be updated to extend the information tracked for each trusted
+person. Currently, each trusted person is represented directly by their public
+key in policy metadata, similar to how TUF represents actors. To support code
+review tool attestations, this must be extended to support the identity of an
+approver from the perspective of the code review tool. For example, Alice may
+use an SSH key for commits and RSL entries but use GitHub's cloud offering for
+code review. Thus, gittuf policy must associate her public key and her GitHub
+username / identifier. Also see: https://github.com/gittuf/gittuf/issues/586.
+
+This extension proposes the following format for the policy schema:
+
+```
+"<id>": {
```

Comment:
what would be the primarily id here?

Reply:
I wanted to leave it up to the users, it could be anything they want? Right now we use keyids but I don't see why we can't have it be usernames or other identifiers based on the context.